resources/patches/scheduler_workbench-allowed_status.patch

```diff
@@ -0,0 +1,197 @@
+     );
+-    drupal_set_message(t('This revision is unpublished and will be published @publish_time.', $replacements), 'status', FALSE);
++    $message = t('This revision is unpublished and will be published @publish_time.', $replacements);
++    drupal_set_message($message, 'status', FALSE);
+   }
+ }
+
+@@ -253,6 +280,23 @@ function _scheduler_workbench_scheduler_unpublish(&$node) {
+  */
+ function _scheduler_workbench_scheduler_pre_publish(&$node) {
+   if ($state = variable_get('scheduler_publish_moderation_state_' . $node->type, FALSE)) {
++    // Get status allowed to be published.
++    $allowed_status = variable_get('scheduler_workbench_allowed_status', array());
++    foreach ($allowed_status as $key => $value) {
++      // If the status is not enabled we remove it from allowed status.
++      if (trim($value) != trim($key)) {
++        unset($allowed_status[$key]);
++      }
++    }
++    // Check if the revision current status allows it to be published.
++    if (!in_array($node->workbench_moderation['current']->state, $allowed_status)) {
++      $options = array(
++        "%node" => $node->nid,
++        '%status' => $node->workbench_moderation['current']->state,
++      );
++      watchdog("scheduler_workbench", t("Node %node with status %status can't be published, check status allowed to be published."), $options, WATCHDOG_INFO);
++      return;
++    }
+     if (!empty($node->workbench_moderation['my_revision']) &&
+       $node->workbench_moderation['my_revision']->state == $state
+     ) {
+@@ -319,6 +363,15 @@ function scheduler_workbench_form_workbench_moderation_moderate_form_alter(&$for
+ function scheduler_workbench_revision_publish($current, $state) {
+   workbench_moderation_moderate($current, $state);
+   $create_publishing_revision = variable_get('scheduler_publish_revision_' . $current->type, 0) == 1;
++  $allowed_status = variable_get('scheduler_workbench_allowed_status', array());
++  if (!in_array($current->workbench_moderation['my_revision']->state, $allowed_status)) {
++    $options = array(
++      "%node" => $current->nid,
++      '%status' => $current->workbench_moderation['my_revision']->state,
++    );
++    watchdog("scheduler_workbench", t("Node %node with status %status, can't be published, check status allowed to be published."), $options, WATCHDOG_INFO);
++    return;
++  }
+   if ($create_publishing_revision) {
+     $current->revision = TRUE;
+     $current->log = t('Saved by Scheduler workbench on @now.', array(
+@@ -348,6 +401,16 @@ function scheduler_workbench_scheduler_allow_publishing($node) {
+     // If this state is one that the current user can access, return TRUE.
+     if ($permitted_states = workbench_moderation_states_next($current_state, $draft_owner, $node)) {
+       $allowed = isset($permitted_states[$published_state]);
++    } // Get status allowed to be published.
++    $allowed_status = variable_get('scheduler_workbench_allowed_status', array());
```

Comment:
Same remark as I made for scheduler_workbench_node_validate(). Maybe encapsulating this variable_get call in its own function could be interesting.

Reply:
Created wrapper _scheduler_workbench_allowed_status()